context/src/main/java/io/grpc/ThreadLocalContextStorage.java

```diff
@@ -28,11 +28,18 @@
   /**
    * Currently bound context.
    */
-  private static final ThreadLocal<Context> localContext = new ThreadLocal<Context>();
+  private static final ThreadLocal<Context> localContext = new ThreadLocal<Context>() {
+    @Override
+    protected Context initialValue() {
+      return Context.ROOT;
```

Comment:
Let's go with option 2, since today implementations are allowed to return null and have `Context` handle it.